uxarray/grid.py

```diff
@@ -0,0 +1,217 @@
+"""uxarray grid module."""
+import os
+import xarray as xr
+import numpy as np
+from warnings import warn
+from pathlib import PurePath
+
+# reader and writer imports
+from ._exodus import read_exodus, write_exodus
+from ._ugrid import read_ugrid, write_ugrid
+from ._shapefile import read_shpfile
+from ._scrip import read_scrip
+from .helpers import determine_file_type
+
+
+class Grid:
+    """The Uxarray Grid object class that describes an unstructured grid.
+
+    Examples
+    ----------
+    Open an exodus file with Uxarray Grid object
+
+    >>> mesh = ux.Grid("filename.g")
+
+    Save as ugrid file
+
+    >>> mesh.saveas("outfile.ug")
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize grid variables, decide if loading happens via file, verts
+        or gridspec If loading from file, initialization happens via the
+        specified file.
+
+        # TODO: Add or remove new Args/kwargs below as this develops further
+
+        Parameters
+        ----------
+        data_arg : string, ndarray, list, tuple, required
```

Comment:
Please put a line-space between doc string sections and arguments as it increases the readability.